infra/src/main/scala/Jaeger.scala

```diff
@@ -0,0 +1,87 @@
+import besom.*
+import besom.util.NonEmptyString
+import besom.api.kubernetes as k8s
+import k8s.core.v1.inputs.*
+import k8s.apps.v1.inputs.*
+import k8s.meta.v1.inputs.*
+import k8s.apps.v1.{DeploymentArgs, deployment}
+import k8s.core.v1.{
+  ConfigMapArgs,
+  Namespace,
+  Service,
+  ServiceArgs,
+  configMap,
+  namespace,
+  persistentVolume,
+  persistentVolumeClaim,
+  service
+}
+import besom.internal.{Context, Output}
+import besom.internal.{Context, Output}
+import besom.util.NotProvided
+
+object Jaeger {
+  val appName = "jaeger"
+  val labels  = Map("app" -> "jaeger")
+  // https://www.jaegertracing.io/docs/1.6/getting-started/#all-in-one-docker-image - port descriptions
+  val ports = Map(
+    "zipkin-thrift-compact" -> (Some("UDP"), 5775),
+    "frontend" -> (None, 16686),
+    "jaeger-thrift-compact" -> (Some("UDP"), 6831),
+    "jaeger-thrift-binary" -> (Some("UDP"), 6832),
+    "configs" -> (None, 5778),
+    "jaeger-thrift-client" -> (None, 14268),
+    "zipkin-collector" -> (None, 9411)
+  )
+  def deploy(using context: Context)(namespace: Namespace) = deployment(
```

Comment:
Besom:

shouldn't we provide some kind of utility of defining service? There is quite a some of boilerplate here?

Reply:
This could be a job for vss maybe? Besom being just scala for pulumi should stick to the pulumi docs. 
Also this could be potentially simplified with helm charts. (I have very little experience with them but they hide some boilerplate.)